configure

```diff
@@ -18,7 +18,17 @@
 ocamlversion=`head -1 VERSION`
 echo "Configuring OCaml version $ocamlversion"
 
-configure_options="$*"
+awk_script='BEGIN {
+   for (i in ARGV) {
+       if (i > 0) {
+          gsub(i, /'\''/, "'\''\\\\'\&\''")
+          printf " '\'%s\''", ARGV[i]
+          delete ARGV[i]
+       }
+   }
+}'
+
+configure_options="$(awk -- "$awk_script" "$@" </dev/null)" || exit 1
```

Comment:
Looks like you're trying to quote the arguments, but:
1. Your script doesn't work (try an argument with a quote on a system with gawk 4.1.1).
2. We don't want to depend on `awk`.
3. It's pointless anyway because `configure_options` is only used for debugging output (modulo your addition of the `make configure` target, a lot of complexity for little benefit IMO).

FWIW, here's my own implementation of a quoting function, without `awk` (using `sed`, which we already depend on):
```
quote1(){ printf "'%s'" "`printf %s "$1" | sed -e "s/'/'\\\\\\\\''/g"`"; }

quote(){
  while : ; do
    case $# in 0) break;; esac
    quote1 "$1"
    shift
    case $# in 0) ;; *) printf " ";; esac
  done
}
```
usage: `quote1 "$1"` or `quote "$@"`